### PR TITLE
Fix #5599: toast doesn't disappear after app exited.

### LINKED ISSUE #5599: BackAndroid.exitApp() prevent to hide Toast message
0.18.0

When I show Toast message and invoke BackAndroid.exitApp() while it still shown, Toast message will not be closed.

```js
ToastAndroid.show('This message will not be closed', ToastAndroid.SHORT);
setTimeout(() => {
  BackAndroid.exitApp();
}, 1000);
```

### PR DESCRIPTION
**Pull request by tdzl2003**
Run `Toast.makeText` other than UI Thread will cause bug #5599 : toast doesn't disappear after app exited.

use `UiThreadUtil.runOnUiThread` to fix this.

### CODE REVIEW

**Comment by tdzl2003** [comment]
As [document saying](https://facebook.github.io/react-native/docs/native-modules-android.html#content), 

> Native modules should not have any assumptions about what thread they are being called on, as the current assignment is subject to change in the future.

**Comment by satya164** [comment]
Thank you for the PR.

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by satya164** [comment]
Have you signed the CLA?

Bot not showing if CLA is signed. cc @bestander @mkonicek

**Comment by satya164** [comment]
I see you have signed the CLA since you've other 13 PRs and many are already accepted. Sorry for the trouble.

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Comment by TanTan-TT** [comment]
@satya164 I believe it is important for users. Could you cherry-pick it into v0.22?

File: ReactAndroid/src/main/java/com/facebook/react/modules/toast/ToastModule.java [hunk 1]
```diff
@@ -46,7 +47,12 @@ public String getName() {
...
   @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
+  public void show(final String message, final int duration) {
+    UiThreadUtil.runOnUiThread(new Runnable() {
+      @Override
+      void run(){
+        Toast.makeText(getReactApplicationContext(), message, duration).show();        
+      }
+    })
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/modules/toast/ToastModule.java]: This does not compile, you're missing a semicolon.

**Comment by tdzl2003** [comment]
sorry for that, I'll fix it now.

**Comment by mkonicek** [comment]
Thanks for the fix @tdzl2003!

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
The code still doesn't compile, please check the CircleCI errors.

**Comment by tdzl2003** [comment]
I updated ToastModule.java now it can compile on my machine. But it still failed on CircleCI due to a npm issue. I don't understand why.

**Comment by bestander** [comment]
@tdzl2003 could you try rebasing on latest master please?
That would most likely fix the issue

**Comment by tdzl2003** [comment]
ESLint failed now, don't know why.

**Comment by bestander** [comment]
Yeah, odd.
It is supposed to lint only files changed in PR.
In this case we can ignore this.
Thanks for looking into it.

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**